opta/cloud/aws/__init__.py

```diff
@@ -0,0 +1,22 @@
+# new-module-api
+
+from opta.cloud.registry import Registry
+from opta.core.terraform2.state import StoreConfig
+
+from .config import AWSProviderConfig
+from .provider import AWSProvider
+from .state import S3Store
+
+__all__ = ["AWSProvider", "AWSProviderConfig", "S3Store"]
+
+
+def register(registry: Registry) -> None:
+    registry.register_provider("aws", AWSProvider, AWSProviderConfig)
+    registry.register_state_store("s3", _make_store, config=AWSProviderConfig)
+    registry.register_default_store("aws", "s3")
```

Comment:
since we will always need all these things we can have one method with mandatory attributes.
`registry.register("aws", AWSProvider, AWSProviderConfig, [store])`
default store could be the first element of the list

Reply:
State storage is a separate concept from which cloud in which terraform is managing resources, so I kept them separate here. It also allows the implementations to be written separately (as I did for `local`, which only has a state storage implementation right now).